__test__/integration/debt_token_api/runners/transfer_from.ts

```diff
@@ -0,0 +1,146 @@
+                }
+
+                const [creditorsTokenID, nonCreditorsTokenID] = tokenIDs;
+
+                // Transfer nonCreditorsToken to a different address
+                await debtTokenAPI.transferFrom(
+                    scenario.creditor,
+                    NON_CREDITOR_ADDRESS,
+                    nonCreditorsTokenID,
+                    "",
+                    { from: scenario.creditor },
+                );
+
+                // Set token approvals
+                await debtTokenAPI.approve(scenario.tokensApprovedOperator, creditorsTokenID, {
+                    from: scenario.creditor,
+                });
+
+                // Set owner's approvals
+                await debtTokenAPI.setApprovalForAll(scenario.ownersApprovedOperator, true);
+
+                scenarioTokenID = scenario.tokenID(
+                    creditorsTokenID,
+                    nonCreditorsTokenID,
+                    NONEXISTENT_TOKEN_ID,
+                );
+
+                // If the recipient is the MockERC721Recipient contract, we have to budget
+                // more gas to account for fees spent on recording the onERC721Received
+                // mock method call.  If the gas is left unspecified (i.e. null), the
+                // debtToken API will choose a default gas value for transferFrom
+                recipientIsContract =
+                    scenarioRecipient === VALID_CONTRACT_RECIPIENT ||
+                    scenarioRecipient === INVALID_CONTRACT_RECIPIENT;
+
+                // HACK: This is a temporary measure for dealing with discrepancies in gas costs
+                //       for different transferFrom calls (depending on how much gas the recipient contract
+                //       guzzles in the onER721Received method call).  A more permanent solution is tracked
+                //       in this Pivotal story: https://www.pivotaltracker.com/story/show/156644350
+                const transferFromGas = recipientIsContract ? 300000 : 200000;
+
+                transferFromPromise = debtTokenAPI.transferFrom(
+                    scenario.from,
+                    scenarioRecipient,
+                    scenarioTokenID,
+                    scenario.data,
+                    {
+                        from: scenario.sender,
+                        gas: transferFromGas,
+                    },
+                );
+
+                if (scenario.succeeds) {
+                    await transferFromPromise;
+                }
+            });
+
+            if (scenario.succeeds) {
+                test(`token's owner has changed to scenario's recipient`, async () => {
+                    await expect(debtTokenAPI.ownerOf(scenarioTokenID)).resolves.toEqual(
+                        scenarioRecipient,
+                    );
+                });
+
+                test("(if recipient is contract) onERC721Received method was called with correct data", async () => {
```

Comment:
shouldn't the if block here be outside the test. why run the test otherwise?

Reply:
good question -- `recipientIsContract` isn't set to its value in the scenario until the `beforeEach` block -- meaning that we can't statically decide whether `recipientIsContract` before runtime.

Reply:
LMK if you have alternative suggestions for how to do this.

Reply:
No, that makes sense to me.